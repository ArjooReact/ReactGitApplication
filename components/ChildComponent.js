import React from 'react'
import {StyleSheet,Button,SafeAreaView,View,FlatList,Text, Alert} from 'react-native'

const ChildComponent = (props)=> {
  // console.log('child component')
  const data = [{
    name: 'Deepak',
    id:11
  },
  {
    name: 'Rajveer_masters',
    id:12
  },
  {
    name: 'Sid_develops',
    id:13
  },
  {
    name: 'Rashmi_masters',
    id:14
  },
  {
    name: 'Rajveer_masters',
    id:12
  },
  {
    name: 'Sid_develops',
    id:13
  },
  {
    name: 'Rashmi_masters',
    id:14
  },{
    name:'Jambo',
    id:14
  }
  ,{
    name:'rambo',
    id:45
  },

  
  
 
]


  // const renderItem = (data)=>{
  //   return<View style= {{width :'100%', height:30, backgroundColor:'green'}}>
  //  <Text style= {{size:20,color:'red'}}>{data.name}</Text>
   
  //   </View>
  // }

  const renderItem = ({ item }) => (
    <View style= {{width :'96%', height:100, backgroundColor:'pink',padding: 10,alignSelf:'center',alignItems:'center', marginVertical: 8,marginHorizontal:6,justifyContent:'center'}}>
   <Text style= {{fontSize:20,color:'red'}}>{item.name}</Text>
   
    </View>
  );
    return <SafeAreaView style = {styles.background}> 
    <View style= {{width:'100%',flex:3,backgroundColor:'green'}}>
    <FlatList
    data = {data}
    renderItem = {renderItem}
    keyExtractor = {(data)=>{data.id}}
    ></FlatList>
    </View>
    <View style= {{width:'100%',flex:1,backgroundColor:'yellow'}}>
      <Button title='PRESS ME' style={{backgroundColor:'blue',width:'100%',height:20}} onPress={()=> props.navigation.navigate('LoginScreen')}>

      </Button>
      <Text>HELLO</Text>
      </View>
   
    
   
    </SafeAreaView>
    

   
}
const styles = StyleSheet.create({
  background:{
    backgroundColor:'green',
      width: '100%',
      height: '100%',
      flex:1,
      flexDirection:'column'
  },
  textColor:{
    color:'green'
  }

})


export default ChildComponent;