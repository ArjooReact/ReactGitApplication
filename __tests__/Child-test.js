import 'react-native'
import React   from 'react'
import ChildComponent from '../components/ChildComponent'
import renderer from 'react-test-renderer'
import { TestScheduler } from 'jest'


test('Childcomponent snapshot',()=>{

    const snap = renderer.create(
        <ChildComponent></ChildComponent>
    ).toJSON()
    expect(snap).toMatchSnapshot();
})