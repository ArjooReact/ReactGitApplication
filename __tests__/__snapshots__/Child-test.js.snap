// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Childcomponent snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "green",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "green",
        "flex": 3,
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "id": 11,
            "name": "Deepak",
          },
          Object {
            "id": 12,
            "name": "Rajveer_masters",
          },
          Object {
            "id": 13,
            "name": "Sid_develops",
          },
          Object {
            "id": 14,
            "name": "Rashmi_masters",
          },
          Object {
            "id": 12,
            "name": "Rajveer_masters",
          },
          Object {
            "id": 13,
            "name": "Sid_develops",
          },
          Object {
            "id": 14,
            "name": "Rashmi_masters",
          },
          Object {
            "id": 14,
            "name": "Jambo",
          },
          Object {
            "id": 45,
            "name": "rambo",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "pink",
                "height": 100,
                "justifyContent": "center",
                "marginHorizontal": 6,
                "marginVertical": 8,
                "padding": 10,
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontSize": 20,
                }
              }
            >
              Deepak
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "pink",
                "height": 100,
                "justifyContent": "center",
                "marginHorizontal": 6,
                "marginVertical": 8,
                "padding": 10,
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontSize": 20,
                }
              }
            >
              Rajveer_masters
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "pink",
                "height": 100,
                "justifyContent": "center",
                "marginHorizontal": 6,
                "marginVertical": 8,
                "padding": 10,
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontSize": 20,
                }
              }
            >
              Sid_develops
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "pink",
                "height": 100,
                "justifyContent": "center",
                "marginHorizontal": 6,
                "marginVertical": 8,
                "padding": 10,
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontSize": 20,
                }
              }
            >
              Rashmi_masters
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "pink",
                "height": 100,
                "justifyContent": "center",
                "marginHorizontal": 6,
                "marginVertical": 8,
                "padding": 10,
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontSize": 20,
                }
              }
            >
              Rajveer_masters
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "pink",
                "height": 100,
                "justifyContent": "center",
                "marginHorizontal": 6,
                "marginVertical": 8,
                "padding": 10,
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontSize": 20,
                }
              }
            >
              Sid_develops
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "pink",
                "height": 100,
                "justifyContent": "center",
                "marginHorizontal": 6,
                "marginVertical": 8,
                "padding": 10,
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontSize": 20,
                }
              }
            >
              Rashmi_masters
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "pink",
                "height": 100,
                "justifyContent": "center",
                "marginHorizontal": 6,
                "marginVertical": 8,
                "padding": 10,
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontSize": 20,
                }
              }
            >
              Jambo
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "pink",
                "height": 100,
                "justifyContent": "center",
                "marginHorizontal": 6,
                "marginVertical": 8,
                "padding": 10,
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontSize": 20,
                }
              }
            >
              rambo
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "yellow",
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          PRESS ME
        </Text>
      </View>
    </View>
    <Text>
      HELLO
    </Text>
  </View>
</RCTSafeAreaView>
`;
